src/ucp/wireup/wireup_cm.c

```diff
@@ -562,21 +552,19 @@ static void ucp_cm_client_connect_cb(uct_ep_h uct_cm_ep, void *arg,
         /* connection can't be established by UCT, no need to disconnect */
         ucp_ep->flags &= ~UCP_EP_FLAG_LOCAL_CONNECTED;
 
-        if ((status == UCS_ERR_NOT_CONNECTED) || (status == UCS_ERR_UNREACHABLE) ||
-            (status == UCS_ERR_CONNECTION_RESET)) {
+        if (((status == UCS_ERR_NOT_CONNECTED) || (status == UCS_ERR_UNREACHABLE) ||
+            (status == UCS_ERR_CONNECTION_RESET)) &&
```

Comment:
can we unite this 'if' with line 551? per https://github.com/openucx/ucx/pull/5551#discussion_r474706191